src/posts/2020-07-17-vim-elixir-ide/index.md

```diff
@@ -0,0 +1,248 @@
+---
+path: /posts/2020-07-17-vim-elixir-ide
+title: Vim as an Elixir IDE
+author: miguel-palhas
+date: 2020-07-15
+tags:
+  - development
+  - elixir
+  - vim
+---
+
+[last-post]: https://subvisual.com/blog/posts/133-super-powered-vim-part-i-projections/
+[vscode]: https://code.visualstudio.com/
+[atom]: https://atom.io/
+[lsp]: https://microsoft.github.io/language-server-protocol/
+[elixir]: https://elixir-lang.org/
+[formatter]: https://hexdocs.pm/mix/1.6.0/Mix.Tasks.Format.html
+[vim-ale]: https://github.com/dense-analysis/ale
+[elixir_ls]: https://github.com/elixir-lsp/elixir-ls
+[quickfix-reflector]: https://github.com/stefandtw/quickfix-reflector.vim
+[vim-projectionist]: https://github.com/tpope/vim-projectionist
+[vim-test]: https://github.com/vim-test/vim-test
+[super-powered-vim]: https://subvisual.com/blog/posts/133-super-powered-vim-part-i-projections/
+[dialixir]: https://github.com/jeremyjh/dialyxir
+[coc-help]: https://github.com/neoclide/coc.nvim/blob/master/doc/coc.txt#L912
+[twitter]: https://twitter.com/naps62
+[dotfiles]: https://github.com/naps62/dotfiles
+
+I would like to talk about how to turn vim into an IDE, with some special focus
+towards [Elixir][elixir], which is the language of choice at Subvisual.
+
+This post is a series of tips on plugins / configs to use in order to make your
+development experience a lot more enjoyable.
+
+## Wait... do I really need Vim?
+
+**No.**
+
+I'm the first one to recommend people not to learn how to use Vim, which
+requires learning a lot of new habits and concepts upfront. Modern editors such
+as [Atom][atom] and [VS Code][vscode] are just as good, or even better, at least
+regarding the learning curve.
+
+In my case, I've been down this rabbit whole for more than a decade. So not only
+have I worked through the learning curve already, I'm also way too deep down the
+rabbit whole at this point. Also, I love it 💚
```

Comment:
I know _exactly_ what you mean by this because I agree with you but reading it rubbed off as pretentious.

The problem with vim is that it's too good but too different. Once you get the hang of it and get past the learning curve, you'll feel the need for those same bindings and thinking patterns in other tools. And you'll end up having to install vims tools like Vimium or being a dork like this:

![image](https://user-images.githubusercontent.com/4325027/87766460-17f2aa80-c811-11ea-9cd0-8bb1fed1db51.png)

I would add something like what I said above because it sounds dismissive of vim and borderline gatekeeping

I would also remove "or even better" because that's subjective, I'd replace with "better suited for your needs".

Reply:
yeah, reading it know (it was late last night) it can sound pretentious yes. I'll try to rephrase this in a bit

> I would also remove "or even better" because that's subjective, I'd replace with "better suited for your needs".

I don't get this part. When saying `just as good, or even better`, both parts are entirely subjective. Not just the "even better" part. "better suited for your needs" is still subjective, just saying the same thing using different words
And I don't have a problem with subjectivity here. This is my opinion, after all. There's no concrete answer on what editor is the best

Reply:
It just rubbed me off weird, as if you were trying too hard to say "hey vim is not that good but I use it" and felt hipster-ish

Reply:
updated